src/libraries/oob-gen.proj

```diff
@@ -1,5 +1,11 @@
 <Project Sdk="Microsoft.Build.Traversal">
 
+  <PropertyGroup Condition="'$(BuildTargetFramework)' != '' and '$(BuildTargetFramework)' == '$(NetFrameworkCurrent)'">
+    <TargetFramework>$(BuildTargetFramework)</TargetFramework>
+    <!-- Filter ProjectReferences to build the best matching target framework only. -->
+    <FilterTraversalProjectReferences>true</FilterTraversalProjectReferences>
```

Comment:
Won't this still overbuild in the case that a generator doesn't multi-target?  What was causing the overbuild to happen?  I would have expected any duplicates to hit the MSBuild project cache - were global properties different?

Reply:
Different global properties, yes. More precisely, the BuildTargetFramework property set.

This solution works for both multi targeting and non multi-targeting builds and is identical to the other oob-*.projs.